core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -177,7 +177,11 @@ public void setCompression(String compressorName) {
     checkState(!sendHeadersCalled, "sendHeaders has been called");
 
     compressor = compressorRegistry.lookupCompressor(compressorName);
-    checkArgument(compressor != null, "Unable to find compressor by name %s", compressorName);
+    if (compressor == null) {
+      throw Status.INTERNAL
```

Comment:
Hmm. The reason I made this change is that the status the client gets back is just UNKNOWN, when really it should be INTERNAL.  The application tried to force the server lib to use a compressor that isn't known.  I wouldn't mind removing the error message, but I would like to keep the code as internal.